recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -34,15 +39,17 @@ this.RecipeRunner = {
     const durabilityManager = new SandboxManager();
     Storage.seedDurability(durabilityManager.sandbox);
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
-    } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Run right now in dev mode
+      this.run();
     }
 
-    setTimeout(this.start.bind(this), delay);
+    this.updateRunInterval();
+    CleanupManager.addCleanupHandler(() => timerManager.unregisterTimer(TIMER_NAME));
+
+    // Watch for the run interval to change, and re-register the timer with the new value
+    prefs.addObserver("run_interval_seconds", this, false);
```

Comment:
The string `run_interval_seconds` is used in multiple places - this should be a `const` declared once instead. Otherwise a typo in one place will break things in a hard-to-detect way.